xarray/core/nanops.py

```diff
@@ -77,15 +77,15 @@ def nanmin(a, axis=None, out=None):
     if a.dtype.kind == "O":
         return _nan_minmax_object("min", dtypes.get_pos_infinity(a.dtype), a, axis)
 
-    module = dask_array if isinstance(a, dask_array_type) else nputils
+    module = dask_array if is_duck_dask_array(a) else nputils
```

Comment:
These nanops will likely require some careful testing, since I'm not sure if using the dask.array module or xarray's nputils is better for non-Dask duck Dask Arrays.